fixtures/log.py

```diff
@@ -41,9 +41,15 @@ def pytest_runtest_logreport(report):
         path, lineno, domaininfo = report.location
         test_status = _test_status(_format_nodeid(report.nodeid, False))
         if test_status == "failed":
-            logger().info(
-                "Managed providers: {}".format(
-                    ", ".join(pytest.store.current_appliance.managed_providers)))
+            try:
+                logger().info(
+                    "Managed providers: {}".format(
+                        ", ".join(pytest.store.current_appliance.managed_providers)))
+            except KeyError as ex:
+                if 'ext_management_systems' in ex.msg:
+                    logger().warning("Unable to query ext_management_systems table; DB issue")
```

Comment:
Yes. During some tests (like migration tests) it can happen that the test fails and we are left with a broken database. In that case, we can't really expect this to work.